File: 25_06_2024_Left_Rotate_Matrix_K_times.java
class Solution {
    public List<List<Integer>> rotateMatrix(int k, List<List<Integer>> mat) {
        int n = mat.size(), m = mat.get(0).size();
        List<List<Integer>> ans = new ArrayList<>(n);
        for (int i = 0; i < n; i++) {
            ans.add(new ArrayList<>(m));
        }
        k %= m;
        if (k == 0) return mat;
        for (int i = 0; i < n; i++) {
            int x = 0;
            for (int j = k; j < m; j++, x++) {
                ans.get(i).set(x, mat.get(i).get(j));
            }
            for (int j = 0; j < k; j++, x++) {
                ans.get(i).set(x, mat.get(i).get(j));
            }
        }
        return ans;
    }
}

File: 25_06_2024_Left_Rotate_Matrix_K_times.py
class Solution:
    def rotateMatrix(self, k: int, mat: List[List[int]]) -> List[List[int]]:
        n, m = len(mat), len(mat[0])
        ans = [[0] * m for _ in range(n)]
        k %= m
        if k == 0:
            return mat
        for i in range(n):
            x = 0
            for j in range(k, m):
                ans[i][x] = mat[i][j]
                x += 1
            for j in range(k):
                ans[i][x] = mat[i][j]
                x += 1
        return ans

